analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -56,7 +56,7 @@ public LexicalAnalyzer(string source)
         }
```

Comment:
> That would remove [S2589](https://sonarsource.github.io/rspec/#/rspec/S2589/csharp) from char.GetUnicodeCategory(c) is var category, that in general we want to avoid.

education: Why we want to avoid this?

Reply:
We use `is var o` only when not using it creates more trouble (that generally means more nesting).
See the forth-last bullet point of [this section of the code conventions](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#code-structure).
The idea is that `is var o` matches every `o`, so, apart from its side-effect of declaring and setting the `o` variable, it behaves like an "always true" condition, which is a bit weird. 
In a sense, it's kind of writing something like the following:
```c
// Online C compiler to run C program online
#include <stdio.h>
#include <stdbool.h>

int main() {
    char c = 'a';
    if (c = '\0', true) {
        printf("c is the string terminator");
    } else {
        printf("c is not the string terminator");
    }
}
```

That prints `c is the string terminator`.
Without that `true`, it would return `c is not the string terminator`.

I needed to give an example in C because that thing doesn't compile in C# :-)

Personally, I don't have any problem with "always true" conditions like `is var o`, and I would use without limitation.